neo/Consensus/RecoveryMessage.cs

```diff
@@ -0,0 +1,109 @@
+        public uint[] ChangeViewTimestamps;
+        public byte[] OriginalChangeViewNumbers;
+
+        // The following 4 fields are to be able to regenerate the PrepareRequest.
+        public UInt256[] TransactionHashes;
+        // The following 3 fields are not serialized if TransactionHashes is null, indicating there is
+        // no PrepareRequest present in the RecoveryMessage.
+        public ulong Nonce;
+        public UInt160 NextConsensus;
+        public MinerTransaction MinerTransaction;
+
+        /// The PreparationHash in case the PrepareRequest hasn't been received yet.
+        /// This can be null if the PrepareRequest information is present, since it can be derived in that case.
+        public UInt256 PreparationHash;
+        public byte[][] PrepareWitnessInvocationScripts;
+        public uint[] PrepareTimestamps;
+        public byte[][] CommitSignatures;
+
+        public RecoveryMessage() : base(ConsensusMessageType.RecoveryMessage)
+        {
+        }
+
+        public override void Deserialize(BinaryReader reader)
+        {
+            base.Deserialize(reader);
+            var txHashCount = reader.ReadVarInt(ushort.MaxValue);
+            if (txHashCount == 0)
+            {
+                TransactionHashes = null;
+                Nonce = 0;
+                NextConsensus = null;
+                MinerTransaction = null;
+            }
+            else
+            {
+                TransactionHashes = new UInt256[txHashCount];
+                for (int i = 0; i < TransactionHashes.Length; i++)
+                {
+                    TransactionHashes[i] = new UInt256();
+                    ((ISerializable) TransactionHashes[i]).Deserialize(reader);
+                }
+                Nonce = reader.ReadUInt64();
+                NextConsensus = reader.ReadSerializable<UInt160>();
+
+                if (TransactionHashes.Distinct().Count() != TransactionHashes.Length)
+                    throw new FormatException();
+                MinerTransaction = reader.ReadSerializable<MinerTransaction>();
+                if (MinerTransaction.Hash != TransactionHashes[0])
+                    throw new FormatException();
+            }
+
+            PreparationHash = reader.ReadVarInt(32) == 0 ? null : reader.ReadSerializable<UInt256>();
+            PrepareWitnessInvocationScripts = reader.ReadVarBytesArray();
+            PrepareTimestamps = reader.ReadUIntArray(PrepareWitnessInvocationScripts.Length);
+
+            ChangeViewWitnessInvocationScripts = reader.ReadVarBytesArray();
+            if (ChangeViewWitnessInvocationScripts != null)
+            {
+                ChangeViewTimestamps = reader.ReadUIntArray(ChangeViewWitnessInvocationScripts.Length);
+                OriginalChangeViewNumbers = reader.ReadVarBytes(255);
+            }
+
+            CommitSignatures = reader.ReadVarBytesArray();
+        }
+
+        public override void Serialize(BinaryWriter writer)
```

Comment:
Is it possible to serialize/deserialize in the following order: `ChangeView` related data, `PrepareRequest` related data, `PrepareResponse` related data, `Commit` related data.

Reply:
Very simple to change. only one the `ChangeView` data needed to be moved. Done.